torch/multiprocessing/reductions.py

```diff
@@ -125,24 +128,63 @@ def reduce_tensor(tensor):
     # *entire* cudaMalloc allocation, i.e., the 0xA000 region, not just
     # the storage 0xA100 (because that is what CUDA supports).  So, on the
     # other end, there simply isn't any way to say, "Wait, you gave me
-    # a bigger region (0xA000) than the one I wanted (0xA100)"; we have
-    # to just make a storage for the entire caching allocator block.
+    # a bigger region (0xA000) than the one I wanted (0xA100)".
+    #
+    # Note that this cudaMalloc allocation might not be a single type of storage.
```

Comment:
I had a little bit of trouble following the line of reasoning in the edited text. Part of the problem is you immediately jump into a paragraph explaining why the cudaMalloc allocation cannot be a single type of storage. OK, so this is definitely something we talked about, but the *reader* of the comment isn't aware of the sordid history of how Edward was silly and tried to mash them all in one storage. You can maybe ease the transition with something like, "OK, so if you sent the cudaMalloc allocation, can you just wrap that up as one storage itself? No, because..."